packages/application-shell/src/components/project-switcher/project-switcher.js

```diff
@@ -2,22 +2,99 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { defaultMemoize } from 'reselect';
 import { graphql } from 'react-apollo';
-import Select from 'react-select';
 import { FormattedMessage, injectIntl } from 'react-intl';
 import flowRight from 'lodash/flowRight';
 import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
 import classnames from 'classnames';
-import { ErrorIcon } from '@commercetools-frontend/ui-kit';
+import { SelectInput, ErrorIcon } from '@commercetools-frontend/ui-kit';
 import styles from './project-switcher.mod.css';
 import ProjectsQuery from './project-switcher.graphql';
 import messages from './messages';
 
-const maxResizableWidth = 225;
+export const ValueContainer = ({ children, projectCount, ...props }) => (
```

Comment:
I think here it makes some sense to destructure `projectCount` because it doesn't need to be passed to the `SelectInput`, and its the easiest way (AFAIK) to omit the prop.

Reply:
I personally don't think using destructuring to omit a bit is explicit at all. It's using a feature to achieve something else. I would have use `omit(props, ['foo'])` to really communicate intend.

Reply:
It's just me though. I think the React community is strangly sold on destrucuring and spreading. Both making code harder to read at times. Given you destructure even `this` and `props` and `state`: how is one supposed to track the origin of something. Left alone the defaulting etc when nesting things. We really had bad experiences with it in the code base.

Reply:
I see, that makes sense to me :) I said that while I was expecting to learn a better way to do it 😄

Reply:
Yes, I myself was used to doing destructuring, mostly because I didn't want to repeat all the `this.props` and `this.state` over and over in the components, and because it's versatile to at the same time omit props from the remaining spread props.
However, I see how that such versatility can actually be a bad thing because it makes it harder to understand why is it being done and, as you said, where those variables are coming from.

So yep, agreed 👍